parsers/gdscript.c

```diff
@@ -0,0 +1,784 @@
+			break;	/* allow 'x = func(b=2,y=2,' lines and comments at the end of line */
+		eq++;
+	}
+
+	/* go backwards to the start of the line, checking we have valid chars */
+	start = cp - 1;
+	while (start >= ln && isspace ((int) *start))
+		--start;
+	while (start >= ln && isIdentifierCharacter ((int) *start))
+		--start;
+	if (!isIdentifierFirstCharacter(*(start + 1)))
+		return NULL;
+	sp = start;
+	while (sp >= ln && isspace ((int) *sp))
+		--sp;
+	if ((sp + 1) != (ln + 3))	/* the line isn't a simple variable assignment */
+		return NULL;
+	/* the line is valid, parse the variable name */
+	++start;
+	return start;
+}
+
+
+/* checks if there is a lambda at position of cp, and return true if so.
+ * We don't return the lambda name since it is useless for now since we already
+ * know it when we call this function, and it would be a little slower. */
+static bool varIsLambda (const char *cp, char **arglist)
+{
+	bool is_lambda = false;
+
+	cp = skipSpace (cp);
+	cp = skipIdentifier (cp); /* skip the lambda's name */
+	cp = skipSpace (cp);
+	if (*cp == '=')
+	{
+		cp++;
+		cp = skipSpace (cp);
+		if (strncmp (cp, "func", 4) == 0)
+		{
+			const char *tmp;
+
+			cp += 4; /* skip the lambda */
+			tmp = skipSpace (cp);
+			/* check if there is a space after lambda to detect assignations
+			 * starting with 'lambdaXXX' */
+			if (tmp != cp)
+			{
+				vString *args = vStringNew ();
+
+				cp = tmp;
+				vStringPut (args, '(');
+				for (; *cp != 0 && *cp != ':'; cp++)
+					vStringPut (args, *cp);
+				vStringPut (args, ')');
+				if (arglist)
+					*arglist = strdup (vStringValue (args));
+				vStringDelete (args);
+				is_lambda = true;
+			}
+		}
+	}
+	return is_lambda;
+}
+
+/* checks if @p cp has keyword @p keyword at the start, and fills @p cp_n with
+ * the position of the next non-whitespace after the keyword */
+static bool matchKeyword (const char *keyword, const char *cp, const char **cp_n)
+{
+	size_t kw_len = strlen (keyword);
+	if (strncmp (cp, keyword, kw_len) == 0 && isspace (cp[kw_len]))
+	{
+		*cp_n = skipSpace (&cp[kw_len + 1]);
+		return true;
+	}
+	return false;
+}
+
+static void findGDScriptTags (void)
+{
+	vString *const continuation = vStringNew ();
+	vString *const name = vStringNew ();
+	vString *const parent = vStringNew();
+
+	NestingLevels *const nesting_levels = nestingLevelsNew(sizeof (struct nestingLevelUserData));
+
+	const char *line;
+	int line_skip = 0;
+	char const *longStringLiteral = NULL;
+
+	while ((line = (const char *) readLineFromInputFile ()) != NULL)
+	{
+		const char *cp = line, *candidate;
+		char const *longstring;
+		char const *keyword, *variable;
+		int indent;
+
+		cp = skipSpace (cp);
+
+		if (*cp == '\0')  /* skip blank line */
+			continue;
+
+		/* Skip comment if we are not inside a multi-line string. */
+		if (*cp == '#' && !longStringLiteral)
+			continue;
+
+		/* Deal with line continuation. */
+		if (!line_skip) vStringClear(continuation);
+		vStringPut(continuation, line);
```

Comment:
As the second argument, VStringPut expects a character, not a C string.
Your compiler may warn something.

Reply:
If you really want to put the C string `line` at the end o continuation, `vStringCatS` may be the function you may want to use.

Reply:
Whoops, they were all vStringCatS before and the CI warning recommended put, wasn't as familiar with the c offerings